diff --git a/htdocs/luci-static/resources/homeproxy.js b/htdocs/luci-static/resources/homeproxy.js
index 369b3a4c01..f5f9990b1f 100644
--- a/htdocs/luci-static/resources/homeproxy.js
+++ b/htdocs/luci-static/resources/homeproxy.js
@@ -304,7 +304,7 @@ return baseclass.extend({
 		if (section_id) {
 			if (!value)
 				return _('Expecting: %s').format(_('non-empty value'));
-			if (ucioption === 'node' && value === 'urltest')
+			if (ucioption === 'node' && ['urltest', 'selector'].includes(value))
 				return true;
 
 			let duplicate = false;
