diff --git a/root/etc/config/homeproxy b/root/etc/config/homeproxy
index 1a53b56e63..e7088fd918 100644
--- a/root/etc/config/homeproxy
+++ b/root/etc/config/homeproxy
@@ -33,6 +33,15 @@ config homeproxy 'config'
 	option ipv6_support '1'
 	option github_token ''
 	option log_level 'warn'
+	option clash_api_enabled '0'
+	option clash_api_external_controller '0.0.0.0:9090'
+	option clash_api_default_mode 'Rule'
+	list clash_api_allow_origin 'https://metacubex.github.io'
+	list clash_api_allow_origin 'https://metacubexd.pages.dev'
+	list clash_api_allow_origin 'http://d.metacubex.one'
+	list clash_api_allow_origin 'https://yacd.metacubex.one'
+	option clash_api_allow_private_network '1'
+	option metacubexd_url 'https://metacubexd.pages.dev/#/overview'
 
 config homeproxy 'control'
 	option lan_proxy_mode 'disabled'
@@ -75,4 +84,3 @@ config homeproxy 'subscription'
 config homeproxy 'server'
 	option enabled '0'
 	option log_level 'warn'
-
