--- --- a/modules/luci-mod-status/htdocs/luci-static/resources/view/status/include/10_system.js	2026-06-08 14:31:37.694020874 +0800
+++ +++ b/modules/luci-mod-status/htdocs/luci-static/resources/view/status/include/10_system.js	2026-06-08 14:32:27.366389253 +0800
@@ -84,7 +84,7 @@
 			_('Model'),            boardinfo.model + cpubench.cpubench,
 			_('Architecture'),     cpuinfo.cpuinfo || boardinfo.system,
 			_('Target Platform'),  (L.isObject(boardinfo.release) ? boardinfo.release.target : ''),
-			_('Firmware Version'), (L.isObject(boardinfo.release) ? boardinfo.release.description + ' / ' : '') + (luciversion || ''),
+			_('Firmware Version'), (L.isObject(boardinfo.release) ? boardinfo.release.description + ' / ' : '') + (luciversion || '') + (' / build by Alex'),
 			_('Kernel Version'),   boardinfo.kernel,
 			_('Local Time'),       datestr,
 			_('Uptime'),           systeminfo.uptime ? '%t'.format(systeminfo.uptime) : null,
