--- --- a/root/usr/share/luci/menu.d/luci-app-bandix.json	2026-06-08 14:31:51.864417301 +0800
+++ +++ b/root/usr/share/luci/menu.d/luci-app-bandix.json	2026-06-08 14:32:27.381597333 +0800
@@ -1,17 +1,17 @@
 {
-	"admin/network/bandix": {
+	"admin/status/bandix": {
 		"title": "Bandix",
-		"order": 90,
+		"order": 8,
 		"action": {
 			"type": "alias",
-			"path": "admin/network/bandix/index"
+			"path": "admin/status/bandix/index"
 		},
 		"depends": {
 			"acl": [ "luci-app-bandix" ],
 			"uci": { "bandix": true }
 		}
 	},
-	"admin/network/bandix/index": {
+	"admin/status/bandix/index": {
 		"title": "Status",
 		"order": 10,
 		"action": {
@@ -19,7 +19,7 @@
 			"path": "bandix/index"
 		}
 	},
-	"admin/network/bandix/connection": {
+	"admin/status/bandix/connection": {
 		"title": "Connections",
 		"order": 15,
 		"action": {
@@ -27,7 +27,7 @@
 			"path": "bandix/connection"
 		}
 	},
-	"admin/network/bandix/dns": {
+	"admin/status/bandix/dns": {
 		"title": "DNS",
 		"order": 20,
 		"action": {
@@ -35,7 +35,7 @@
 			"path": "bandix/dns"
 		}
 	},
-	"admin/network/bandix/settings": {
+	"admin/status/bandix/settings": {
 		"title": "Settings",
 		"order": 25,
 		"action": {
