--- --- a/files/etc/init.d/pbr	2026-06-08 14:32:07.907089512 +0800
+++ +++ b/files/etc/init.d/pbr	2026-06-08 14:32:27.389201373 +0800
@@ -568,7 +568,7 @@
 print_json_string() { json_init; json_add_string "$1" "$2"; json_dump; json_cleanup; }
 
 stop_forward() {
-	if [ -n "$strict_enforcement" ] && [ "$(cat /proc/sys/net/ipv4/ip_forward)" != "0" ]; then
+	if [ -n "$enabled" ] && [ -n "$strict_enforcement" ] && [ "$(cat /proc/sys/net/ipv4/ip_forward)" != "0" ]; then
 		/sbin/sysctl -w net.ipv4.ip_forward=0 >/dev/null 2>&1
 		/sbin/sysctl -w net.ipv6.conf.all.forwarding=0 >/dev/null 2>&1
 		output "Forwarding is disabled\n"
